### PR TITLE
[FOGL-3135]: Added modal for adding and enable/disable Notification service

### PR DESCRIPTION
**Pull request by MonikaSharma06**
Removed `disabled now` alert from Notification page (screenshot is from develop branch)-
notification-develop

Initial view of Notification Page-
notification-initialView

Notification modal (click on setting icon)-
notification-modal-1

After adding Notification service-
notification-enabled

Notification modal (click on setting icon), user can enable/disable and delete notification from here-
notification-modal-toUpdate

After disabling the notification-
notification-disabled

### CODE REVIEW

File: src/app/components/core/notifications/notification-service-modal/notification-service-modal.component.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<div id="notification-setting-modal" class="modal">
```
Comment (shariq1501) [inline src/app/components/core/notifications/notification-service-modal/notification-service-modal.component.html]: Rename  `notification-setting-modal` -> `notification-service-modal` for naming consistency.

File: src/app/components/core/notifications/notifications.component.ts [hunk 2]
```diff
@@ -370,6 +245,18 @@ export class NotificationsComponent implements OnInit, OnDestroy {
         });
   }
 
+  /**
+   * Open Notification Settings modal
+   */
+  openNotificationSettingsModal() {
```
Comment (shariq1501) [inline src/app/components/core/notifications/notifications.component.ts]: `openNotificationSettingsModal -> openNotificationServiceModal`

File: src/app/components/core/notifications/notifications.component.ts [hunk 3]
```diff
@@ -370,6 +245,18 @@ export class NotificationsComponent implements OnInit, OnDestroy {
...
+    this.childData = {
+      'isNotificationServiceAvailable': this.isNotificationServiceAvailable,
```
Comment (shariq1501) [inline src/app/components/core/notifications/notifications.component.ts]: IMO in an object key holds the information. key name  `isNotificationServiceAvailable -> notificationServiceAvailable`. remove questionable `is`  from all keys. Also there is no need of single quote as well.

File: src/app/components/core/notifications/notification-service-modal/notification-service-modal.component.ts [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+  public category: any;
+  public useProxy: 'true';
+  isNotificationServiceAvailable = false;
+  isNotificationServiceEnabled = false;
+  notificationServiceName = '';
+  public changedChildConfig = [];
+  public availableServices = [];
+  notificationServicePackageName = 'foglamp-service-notification';
+
+  @Output() notifySettingEmitter: EventEmitter<any> = new EventEmitter<any>();
```
Comment (shariq1501) [inline src/app/components/core/notifications/notification-service-modal/notification-service-modal.component.ts]: rename  notifySettingEmitter -> notifyServiceEmitter

File: src/app/components/core/notifications/notification-service-modal/notification-service-modal.component.ts [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+    }
+    this.name = this.notificationServiceName;
+    this.enabled = this.isNotificationServiceEnabled;
+    if (this.notificationServiceName) {
+      this.getCategory();
+    }
+  }
+
+  public toggleModal(isOpen: Boolean) {
+    const notification_setting_modal = <HTMLDivElement>document.getElementById('notification-setting-modal');
```
Comment (shariq1501) [inline src/app/components/core/notifications/notification-service-modal/notification-service-modal.component.ts]: `notification-setting-modal -> notification-service-modal`

**Review by shariq1501 — CHANGES_REQUESTED** [review]

File: src/app/components/core/notifications/notification-service-modal/notification-service-modal.component.html [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+                        <div class="field is-horizontal">
+                            <div class="field-label has-text-left">
+                                <label class="label">Name</label>
+                            </div>
+                            <div class="field-body">
+                                <div class="field">
+                                    <input *ngIf="isNotificationServiceAvailable" type="text" class="input is-small" disabled name="name" required trim="blur" [(ngModel)]="name" />
+                                    <input *ngIf="!isNotificationServiceAvailable" type="text" class="input is-small" name="name" [(ngModel)]="name" placeholder="FogLAMP Notifications"
+                                    required trim="blur" />
+                                    <small *ngIf="fg?.controls['name']?.invalid" class="help is-danger level-left">Name is required</small>
```
Comment (shariq1501) [inline src/app/components/core/notifications/notification-service-modal/notification-service-modal.component.html]: add `*` before `Name is required`

File: src/app/components/core/notifications/notification-service-modal/notification-service-modal.component.ts [hunk 7]
```diff
@@ -0,0 +1,353 @@
...
+import { isEmpty } from 'lodash';
+
+@Component({
+  selector: 'app-notification-service-modal',
+  templateUrl: './notification-service-modal.component.html',
+  styleUrls: ['./notification-service-modal.component.css']
+})
+export class NotificationServiceModalComponent implements OnChanges {
+  enabled: Boolean;
+  name: string;
```
Comment (shariq1501) [inline src/app/components/core/notifications/notification-service-modal/notification-service-modal.component.ts]: this name variable is not required we already have `notificationServiceName` variable that holds the service name. We can pass  `notificationServiceName` in html file in input field `[(ngModel)]='notificationServiceName'`.

**Review by shariq1501 — CHANGES_REQUESTED** [review]

File: src/app/components/core/notifications/notification-service-modal/notification-service-modal.component.html [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+                        <div class="field is-horizontal">
+                            <div class="field-label has-text-left">
+                                <label class="label">Name</label>
+                            </div>
+                            <div class="field-body">
+                                <div class="field">
+                                    <input *ngIf="!isNotificationServiceAvailable" type="text" class="input is-small"
+                                        name="name" [(ngModel)]="name" placeholder="FogLAMP Notifications" required
+                                        trim="blur" />
+                                    <small *ngIf="fg?.controls['name']?.invalid"
```
Comment (shariq1501) [inline src/app/components/core/notifications/notification-service-modal/notification-service-modal.component.html]: `<small *ngIf="fg?.controls['name']?.dirty && fg?.controls['name']?.invalid "
                                        class="help is-danger level-left">*required</small>` to hide required field check on page load.

Comment (MonikaSharma06) [reply src/app/components/core/notifications/notification-service-modal/notification-service-modal.component.html]: It doesn't work, if form is submitted with blank input value of name (without touching name field)

Comment (shariq1501) [reply src/app/components/core/notifications/notification-service-modal/notification-service-modal.component.html]: `if (!this.form.valid) {
      this.form.controls['name'].markAsTouched();
      return;
    }` 
use this check inside proxy() method to fix (without touching the field issue)

**Review by shariq1501 — APPROVED** [review]

**Review by praveen-garg — APPROVED** [review]
lgtm!

**Comment by billehunt** [comment]
The ideal behavior is to remove the config and enable/disable completely from this page, and instead have it inside the configuration tab. However, I assume this isn't possible currently. If that's the case, please merge this PR, and we'll add it to the main config later. (not a high priority).

**Pull request merged**